### PR TITLE
Dealii, Versions added

### PR DESCRIPTION
**Pull request by ashkan2200**
Included additional versions for Deal.II package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/dealii/package.py [hunk 1]
```diff
@@ -30,9 +30,13 @@ class Dealii(Package):
     """C++ software library providing well-documented tools to build finite
     element codes for a broad variety of PDEs."""
     homepage = "https://www.dealii.org"
-    url      = "https://github.com/dealii/dealii/releases/download/v8.4.0/dealii-8.4.0.tar.gz"
+    url      = "https://github.com/dealii/dealii/releases/download/v8.4.1/dealii-8.4.1.tar.gz"
```
Comment (davydden) [inline var/spack/repos/builtin/packages/dealii/package.py]: you don't need to change that line. Actual URL will be picked up automatically.

Comment (ashkan2200) [inline var/spack/repos/builtin/packages/dealii/package.py]: @davydden true but just wanted to keep that line up to date as well. Would it make a difference that I have changed it?

Comment (davydden) [inline var/spack/repos/builtin/packages/dealii/package.py]: it should not. 

btw, did you test that URL for versions is picked up? With you patch, did you try to install, say, `8.4.0` and `8.3.0`?

Comment (ashkan2200) [inline var/spack/repos/builtin/packages/dealii/package.py]: I tried 8.3.0 and it picked the version correctly and downloaded the file. I halted the compilation though as it takes a long time to finish.

Comment (ashkan2200) [inline var/spack/repos/builtin/packages/dealii/package.py]: I checked that all the versions are downloaded/staged correctly.

Comment (davydden) [inline var/spack/repos/builtin/packages/dealii/package.py]: nice

**Comment by ashkan2200** [comment]
@davydden Now I tried to install 8.3.0 it compiles but fails on the example step-40. I checked and it is because step-40 is changed from the previous version. If I comment the whole step-# compilation then the package builds correctly and it passes all the unit tests too.

Why did you put these tests? Isn't it enough that you are doing make test? 
One option is to make the tests version specific.

**Comment by davydden** [comment]
@ashkan2200 no, it is not enough just to run make test because very few things are tested in `make test`. The proper test suite is about 8000 unit tests and takes like 20G of harddisk space and some hours of time. So those extra tests I added  based on examples are there to cover basic third-part-libraries like Mumps, Superlu_dist, PETSc, Trilinos, OpenCascade, Slepc.

I would suggest you to wrap offending part in the if statment
```
if version > Version('8.x.y')

```
I suppose you should be able to run Step-40, but probably should not try to modify preconditioners/solver to be used.

**Pull request merged**